src/config/syscheck-config.c

```diff
@@ -947,9 +947,14 @@ static int read_attr(syscheck_config *syscheck, const char *dirs, char **g_attrs
                 strchr(real_path, '?') ||
                 strchr(real_path, '[')) {
             int gindex = 0;
+            int gstatus;
             glob_t g;
 
-            if (glob(tmp_dir, 0, NULL, &g) != 0) {
+            if (gstatus = glob(tmp_dir, 0, NULL, &g), gstatus == GLOB_NOMATCH) {
+                mdebug2(GLOB_NO_MATCH, real_path);
+                dir++;
+                continue;
+            } else if (gstatus = GLOB_NOSPACE | GLOB_ABORTED) {
```

Comment:
This line produces a warning:
```
config/syscheck-config.c: In function ‘read_attr’:
config/syscheck-config.c:957:24: warning: suggest parentheses around assignment used as truth value [-Wparentheses]
  957 |             } else if (gstatus = GLOB_NOSPACE | GLOB_ABORTED) {
```

```suggestion
            } else {
```